### DISCUSSION TITLE
How do you handle CORS with TypeScript based Server

### DISCUSSION THREAD

**Discussion by radiantone** [socketio/socket.io#4327] [General]
Hi,
  I saw the docs for how to handle CORS in socket.io server, but it doesn't show how to do this for typescript.

Thanks!

**Comment by darrachequesne** [comment]
Hi!

You can use the exact same set of options with TypeScript:

```ts
import { createServer } from "http";
import { Server } from "socket.io";

const httpServer = createServer();
const io = new Server(httpServer, {
  cors: {
    origin: "https://example.com"
  }
});
```

Reference: https://socket.io/docs/v4/handling-cors/
